### ISSUE TITLE
No datastreams are listed on Fleet

### ISSUE THREAD

**Issue by mdelapenya** [elastic/e2e-testing#1274]
The step `system package dashboards are listed in Fleet ` is failing because it does not faind any data stream in the max timeout (3min).

It fails on Centos and Debian, in both AMD and ARM.

### Steps to reproduce
1 Run:
```shell
$ TAGS="fleet_mode_agent && install && centos" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE DEVELOPER_MODE=true ELASTIC_APM_ACTIVE=false make -C e2e/_suites/fleet functional-test
```

**Expected behaviour:** the scenario passes
**Current behaviour:**
--- Failed steps:

```
  Scenario Outline: Deploying the centos agent # features/fleet_mode_agent.feature:6
    And system package dashboards are listed in Fleet # features/fleet_mode_agent.feature:10
      Error: There are no datastreams yet


1 scenarios (1 failed)
4 steps (3 passed, 1 failed)
5m22.255990276s
make: *** [functional-test] Error 1
```

**Comment by mdelapenya** [comment]
I'm currently bisecting the test execution between two kibana commits.

I've built and pushed Kibana images for those commits, only for AMD, as the APM-CI job is only building the AMD image. For that reason, CI builds below will contain errors for all ARM stages. But we do not care, as we want to verify if the image breaks the tests, never mind whether it is AMD or ARM.

### 1st commit: 4a941565502547f96bab72786e1ac11f61f19558 (elastic/kibana#101828)
- CI build: https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp/detail/master/1002/pipeline/

### 2nd commit: cd5cd65fb2ec04ed63fcbc6b87f1fdb7333bee72 (elastic/kibana#102219)
-  CI build: https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp/detail/master/1003/pipeline/

Will post results here.

**Comment by mdelapenya** [comment]
It's weird: I've tested with an old Kibana image (`pr101655`, from June 8th 2021) and the test fails. I think the error is not on Kibana but in the other pieces: fleet-server or the agent. I cannot run the tests for incremental commits of the agent because the artifacts are not generated in the GCP bucket.

**Comment by mdelapenya** [comment]
Mmm, trying previous step with a clean environment, removing all services in the compose file, and now fleet-server starts properly.

But unfortunately the revoke test still fails:

```
--- Failed steps:

  Scenario Outline: Revoking the enrollment token for the centos agent # features/fleet_mode_agent.feature:105
    Then an attempt to enroll a new agent fails # features/fleet_mode_agent.feature:108
      Error: The agent was enrolled although the token was previously revoked
```

So if the fleet-server is from 8 days ago, when the tests were supposed to be passing, and the tests actually fail, I'd say it's because of another piece of the stack: it seems it's not kibana, it seems it's not fleet-server. Let's check with the agent. I'm gonna bisect the agent, although I'm seeing problems with the packaging job not producing the `commits` artifacts for all commits cc/ @elastic/observablt-robots

**Comment by adam-stokes** [comment]
Could this be an elasticsearch change? Maybe the query has changed?

**Comment by adam-stokes** [comment]
This is due to the way we currently bring up Kibana, the environment variable `XPACK_FLEET_AGENTS_FLEET_SERVER_HOSTS` is not being honored properly it seems.  Same reason why #1273 is failing as well. Will post a reference bug once available

**Comment by cachedout** [comment]
@adam-stokes and @mdelapenya did https://github.com/elastic/e2e-testing/pull/1281 fix this as well?

**Comment by mdelapenya** [comment]
> @adam-stokes and @mdelapenya did #1281 fix this as well?

No, that was not a solution and this issue is still under investigation. We'll post here more work about it

**Comment by EricDavisX** [comment]
I logged a product team bug for this, thanks so much for noting it Manu in slack.  def sounds like the same issue:
https://github.com/elastic/beats/issues/26518
 - this has some good research in it, hopefully helpful to the team!

**Comment by adam-stokes** [comment]
This looks to have been fixed, will wait for @mdelapenya to verify but our tests are passing again

**Comment by EricDavisX** [comment]
the duplicate issue was re-tested and found fixed - let's not wait, i'm closing it out.  there are other reasons for the tests to fail, if they still are, and we should do new tickets.  : /  thanks Adam.

**Comment by mdelapenya** [comment]
Fixed in https://github.com/elastic/kibana/pull/104415
